wasm/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+use css_inline as rust_inline;
+use wasm_bindgen::prelude::*;
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+
+struct InlineErrorWrapper(rust_inline::InlineError);
+
+impl From<InlineErrorWrapper> for JsValue {
+    fn from(error: InlineErrorWrapper) -> Self {
+        JsValue::from_str(error.0.to_string().as_str())
+    }
+}
+
+struct UrlError(url::ParseError);
+
+impl From<UrlError> for JsValue {
+    fn from(error: UrlError) -> Self {
+        wasm_bindgen::throw_str(&error.0.to_string())
+    }
+}
+
+fn parse_url(url: Option<String>) -> Result<Option<url::Url>, JsValue> {
+    Ok(if let Some(url) = url {
+        Some(url::Url::parse(url.as_str()).map_err(UrlError)?)
+    } else {
+        None
+    })
+}
+
+#[macro_use]
+extern crate serde_derive;
+
+#[derive(Deserialize)]
+#[serde(default)]
+pub struct Options {
```

Comment:
This struct could be private (I missed it originally), I propose to remove `pub`
```suggestion
struct Options {
```